### DISCUSSION TITLE
Using the redirect component

### DISCUSSION THREAD

**Discussion by lozdown** [sqlpage/SQLPage#194] [Q&A] [answered]
@lovasoa  Thank you for SQLPage. It is a very interesting concept and great software!  
I am wanting to understand how best to use the **redirect** component.  Particularly I want a tidy way to confirm delete operations. As **redirect** has to be the first statement I have had to use two files, 'delete_user.sql'  for user to confirm delete, the second 'delete_confirm.sql' to execute the delete, each with parameter $user_id to identify the row to delete.  

Is there a way to combine them in one file and achieve the same effect?

# URL: delete_user.sql?user_id=1

```sql
select 'alert' as component,
	'Deleting id ' || coalesce($user_id,'-') as title;
select 'Continue' as title, 'delete_confirm.sql'|| COALESCE('?id=' || $user_id, '') as link; 
select 'Cancel' as title, 'index.sql' as link;
```
# URL: delete_confirm.sql?user_id=1

```sql
DELETE FROM "user"
	WHERE id=$id::int
returning 'redirect' AS component, 
'index.sql' as link;	
```

**Comment by lovasoa** [comment, marked as answer]
The way you are doing it now is, I think, simple and idiomatic. If you really want to merge this into a single file, you can add a URL parameter:


```sql
DELETE FROM "user" WHERE id=$user_id::int AND $confirmed IS NOT NULL
RETURNING 'redirect' AS component, 'index.sql' as link;

select 'alert' as component, 'Deleting id ' || $user_id as title;
select 'Continue' as title, '?confirmed&user_id=' || $user_id as link; 
select 'Cancel' as title, 'index.sql' as link;

**Reply by lozdown** [reply]
Perfect, I would not have thought of that! - Thank you
